app/src/main/java/helium314/keyboard/latin/inputlogic/InputLogic.java

```diff
@@ -1782,9 +1816,19 @@ private void restartSuggestions(final TextRange range) {
         }
     }
 
-    void doShowSuggestionsAndClearAutoCorrectionIndicator(final SuggestedWords suggestedWords) {
+    private void doShowSuggestionsAndClearAutoCorrectionIndicator(final SuggestedWords suggestedWords) {
         mIsAutoCorrectionIndicatorOn = false;
-        mLatinIME.mHandler.showSuggestionStrip(suggestedWords);
+        if (suggestedWords != null) {
+            mSuggestionStripViewAccessor.setSuggestions(suggestedWords);
+            if (shouldAutoShowSuggestions(Settings.getValues(), suggestedWords)) {
+                mSuggestionStripViewAccessor.showSuggestionStrip();
+            }
+        }
+    }
+
+    private boolean shouldAutoShowSuggestions(SettingsValues settingsValues, SuggestedWords suggestedWords) {
+        return ! suggestedWords.isEmpty() && settingsValues.isSuggestionsEnabledPerUserSettings() && ! mWordComposer.isResumed()
+                    && isEmojiSearch();
```

Comment:
This is confusing. Why should suggestions be shown automatically only if it's an emoji search?

Reply:
I think what had me confused is the name. `shouldAutoShowSuggestions` seems to be used to determine whether the suggestion _strip_ is shown.
Since it's used the same way in both uses, maybe the check and action could be merged into some `showSuggestionStripIfEmojiSearch`.